src/components/Tables/AppletsTable/Table/AppletItem/ShareApplet/ShareApplet.tsx

```diff
@@ -0,0 +1,326 @@
+      setRemovedFromLibrary(true);
+      setSecondBtnVisible(false);
+      setMainBtnText(t('ok'));
+      setTitle(t('appletIsRemovedFromLibrary', { appletName: applet.name }) || '');
+
+      dispatch(
+        folders.actions.updateAppletData({
+          appletId: applet.id,
+          published: false,
+        }),
+      );
+    }
+  };
+
+  const handleUpdateSharedApplet = async () => {
+    setIsLoading(true);
+    const changeNameResult = await handleChangeName();
+    const updateTermsResult = await handleUpdateSearchTerms();
+    const libraryUrlResult = await getAppletLibUrl();
+
+    if (
+      changeNameResult.status === 200 &&
+      updateTermsResult.status === 200 &&
+      libraryUrlResult.status === 200
+    ) {
+      setIsLoading(false);
+      setLibraryUrl(libraryUrlResult.data || '');
+      setAppletUpdated(true);
+
+      dispatch(
+        folders.actions.updateAppletData({
+          appletId: applet.id,
+          appletName: getValues().appletName,
+          published: true,
+        }),
+      );
+    }
+  };
+
+  const handleAddKeyword = (value: string) => {
+    if (value.length > 0) {
+      setValue('keyword', '');
+      setKeywords((prevState) => {
+        if (prevState?.some((item) => item === value)) {
+          return prevState;
+        }
+        const res = prevState?.concat(value);
+        setValue('keywords', res);
+
+        return res;
+      });
+    }
+  };
+
+  const handleRemoveKeyword = (index: number) => {
+    setKeywords((prevState) => {
+      const res = prevState?.filter((_, i) => i !== index);
+      setValue('keywords', res);
+
+      return res;
+    });
+  };
+
+  const hasError =
+    checkNameError || changeNameError || publishError || getLibraryUrlError || updateTermsError;
+
+  useEffect(() => {
+    if (applet.published && !removedFromLibrary) {
+      (async () => {
+        const { getAppletSearchTerms } = folders.thunk;
+        const result = await dispatch(getAppletSearchTerms({ appletId: applet.id }));
+
+        if (getAppletSearchTerms.fulfilled.match(result)) {
+          setKeywords(result.payload.data.keywords);
+        }
+      })();
+
+      setTitle(t('updateSharedApplet') || '');
+      setMainBtnText(t('stopSharing'));
+      setSecondBtnVisible(true);
+    }
+
+    if (appletShared) {
+      setTitle(t('appletIsSharedWithLibrary') || '');
+      setSecondBtnVisible(false);
+      setMainBtnText(t('ok'));
+    }
+
+    if (appletUpdated) {
+      setTitle(t('appletIsUpdatedSuccessfully') || '');
+      setSecondBtnVisible(false);
+      setMainBtnText(t('ok'));
+    }
+  }, [applet, appletShared, appletUpdated, removedFromLibrary, dispatch, t]);
+
+  let modalComponent: JSX.Element | null = (
+    <>
+      {isLoading && <Spinner />}
+      <StyledInputWrapper>
+        <InputController
+          fullWidth
+          name="appletName"
+          error={showNameTakenError}
+          control={control}
+          label={`${t('appletName')}*`}
```

Comment:
I would not add `*` to the label, I suppose you can add `require` prop for InputController?

Reply:
done